src/index/base.cpp

```diff
@@ -72,12 +78,30 @@ void BaseIndexNotifications::blockConnected(const interfaces::BlockInfo& block)
         return;
     }
 
+    CBlockUndo block_undo;
+    if (m_options.connect_undo_data && !block.undo_data && pindex->nHeight > 0) {
+        if (!m_index.m_chainstate->m_blockman.UndoReadFromDisk(block_undo, *pindex)) {
+            m_index.FatalErrorf("%s: Failed to read block %s undo data from disk",
+                       __func__, pindex->GetBlockHash().ToString());
+            return;
+        }
+        block.undo_data = &block_undo;
+    }
+
     if (!m_index.CustomAppend(block)) {
         m_index.FatalErrorf("%s: Failed to write block %s to index",
                    __func__, pindex->GetBlockHash().ToString());
         return;
     }
 
+    if (!m_index.m_synced) {
```

Comment:
In 5e2b1ea1:

I'm wondering what would change if we use `if (!block.chain_tip)` instead. Guess that nothing right?